compiler/src/iree/compiler/GlobalOptimization/UninitializedValueValidation.cpp

```diff
@@ -0,0 +1,57 @@
+// Copyright 2025 The IREE Authors
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include "iree/compiler/GlobalOptimization/Passes.h"
+#include "llvm/ADT/SmallSet.h"
+#include "llvm/Support/Debug.h"
+#include "mlir/Dialect/Linalg/IR/Linalg.h"
+#include "mlir/Dialect/Tensor/IR/Tensor.h"
+#include "mlir/Transforms/GreedyPatternRewriteDriver.h"
+
+namespace mlir::iree_compiler::GlobalOptimization {
+
+#define GEN_PASS_DEF_UNINITIALIZEDVALUEVALIDATIONPASS
+#include "iree/compiler/GlobalOptimization/Passes.h.inc"
+
+namespace {
+
+class UninitializedValueValidationPass final
+    : public impl::UninitializedValueValidationPassBase<
+          UninitializedValueValidationPass> {
+public:
+  void runOnOperation() override {
+    Operation *op = getOperation();
+    op->walk([&](linalg::LinalgOp linalgOp) -> WalkResult {
```

Comment:
I dont think a blanket check on `linalgOp` works. This is invalid only if the `linalgOp` has reduction iterator types.